code/modules/multiz/movement.dm

```diff
@@ -215,54 +215,58 @@
 
 /mob/living/handle_fall(var/turf/landing)
 	if(..())
-		return
+		return 1
 
 	var/area/area1 = get_area(landing)
 	if(!area1.has_gravity())
-		return
+		return 1
 
 	if(istype(landing, /turf/simulated/open))
-		var/turf/simulated/open/open = landing
-		var/area/area2 = get_area(open.below)
-		if(!area2.has_gravity())
-			return
+		z_levels_fallen++ // Don't damage them but keep track of this.
+		return 1
+
+	if(istype(landing, /turf/space))
+		z_levels_fallen = 0 // turns out they didn't hit anything solid. Lucky them.
+		return 1
+
+	if(!z_levels_fallen) // Checks if they only fell down one floor.
+		z_levels_fallen = 1
 
 	if(!istype(src, /mob/living/carbon/human))
 		var/damage = 30
-		apply_damage(rand(0, damage), BRUTE)
-		apply_damage(rand(0, damage), BRUTE)
-		apply_damage(rand(0, damage), BRUTE)
+		apply_damage(rand(0, damage*z_levels_fallen), BRUTE)
+		apply_damage(rand(0, damage*z_levels_fallen), BRUTE)
+		apply_damage(rand(0, damage*z_levels_fallen), BRUTE)
 
 /mob/living/carbon/human/handle_fall(var/turf/landing)
 	if(..())
 		return
-
 	var/damage = 30*species.fall_mod
```

Comment:
What Fowl said. After this line, you can do `damage *= z_levels_fallen` and just use `damage` from here on out.